### PR TITLE
MessageBar: Fix high contrast mode issue where you cannot tell which buttons you have focus on.

### LINKED ISSUE #9934: Focusing on MessageBar buttons in high contrast isn't visible
In high contrast when a button of the `MessageBar` is focused, it isn't obvious what element is focused.

We are using OUFR v6.

Impacted products: AI Builder in Powerapps

### PR DESCRIPTION
**Pull request by cliffkoh**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9934
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Fix high contrast bugs in message bar. Also made high contrast focus rects for dismiss/expand button consistent.

##### Before: 
https://fabricweb.z5.web.core.windows.net/oufr/7.16.0/#/examples/messagebar

##### After:
http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/9953/merge/fabric-website-resources/dist/index.html#/examples/messagebar

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9953)

### CODE REVIEW

**Comment by cliffkoh** [comment]
@msft-github-bot please merge this once @khmakoto and @ecraig12345 has graciously given their sign off

File: packages/office-ui-fabric-react/src/components/Button/MessageBarButton/MessageBarButton.styles.ts [hunk 1]
```diff
@@ -2,16 +2,27 @@ import { IButtonStyles } from '../Button.types';
 import { ITheme, concatStyleSets } from '../../../Styling';
 import { memoizeFunction } from '../../../Utilities';
 import { getStyles as getBaseButtonStyles } from '../BaseButton.styles';
+import { getFocusStyle } from '@uifabric/styling';
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Button/MessageBarButton/MessageBarButton.styles.ts]: This import can and should be done in line 2 import.

File: packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts [hunk 2]
```diff
@@ -84,21 +85,31 @@ export const getStyles = (props: IMessageBarStyleProps): IMessageBarStyles => {
...
+      borderColor: 'transparent'
+    }),
+    {
+      flexShrink: 0,
+      width: 32,
+      height: 32,
+      padding: '8px 12px',
+      selectors: {
+        '& .ms-Button-icon': dismissalAndExpandIconStyle,
+        ':hover': {
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts]: I know this part is just moved but this selector could be combined with the next (`:hover, :active`)

**Comment by Vitalius1** [comment]
Interesting... when I go to try out HC mode on our live website I can see what button I focus bun in a not very useful way 😄 
messagebar

... so I wonder why I can see a focus happening on the live website and not on the [deploy link](https://fabricweb.z5.web.core.windows.net/oufr/7.16.0/#/examples/messagebar) @cliffkoh mentioned in the issue.

**Comment by cliffkoh** [comment]
Live website has styles from the surrounding page loaded too, which may have caused the behavior you observed. Hence I typically use aka.ms/fabricdemo.

**Comment by cliffkoh** [comment]
BTW the "Before" link is not a deploy link but a link from aka.ms/fabricdemo

**Comment by Vitalius1** [comment]
Aren’t deploy links about the same as fabric demo ones?

**Comment by ecraig12345** [comment]
With deploy links you have the choice of the demo site or the full website.

**Comment by cliffkoh** [comment]
@Vitalius1 depends on what you mean by same but technically they're quite different. PR deploy links includes the demo pages for other packages as well - the fabricdemo ones are limited only to the main OUFR package (or fabric-website-resources package), has a different URL, and is limited to the demo site (not the full website).

**Comment by Vitalius1** [comment]
Not taking the URL into discussion how different is [this link](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/9953/merge/fabric-website-resources/dist/index.html) which is the `fabric-website-resources` version of the deploy links and [this link](https://fabricweb.z5.web.core.windows.net/oufr/7.16.0/#/examples/messagebar) which is the `aka.ms/fabricdemo` link?

**Comment by cliffkoh** [comment]
They're the same thing (ignoring version differences). Both are fabric-website-resources.

File: packages/office-ui-fabric-react/src/components/Button/MessageBarButton/MessageBarButton.styles.ts (hunk 1 above)
Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Button/MessageBarButton/MessageBarButton.styles.ts]: Thanks for catching :) The import was added by vscode, I believe.

File: packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts (hunk 2 above)
Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts]: Combining them however has the adverse effect of making it harder for concatStyleSets to kick in, therefore making it harder to customize/overwrite these selectorts

**Comment by ecraig12345** [comment]
@msft-github-bot Make sure @betrue-final-final approves

File: packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts (hunk 2 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.styles.ts]: Oh interesting...mergeStyles is so hard...

**Comment by betrue-final-final** [comment]
Why is the focus rect outside the button in Standard, and invisible in Primary? I'm looking at the Deploy link. in the checks.
image

**Comment by betrue-final-final** [comment]
Maybe I'm not looking at the right link?

**Comment by betrue-final-final** [comment]
image
"Demo site" right?

**Comment by cliffkoh** [comment]
It looks like you're looking into Button, this PR changes MessageBar only.

Specifically compare and contrast

Before:
https://fabricweb.z5.web.core.windows.net/oufr/7.16.0/#/examples/messagebar

After:
http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/9953/merge/fabric-website-resources/dist/index.html#/examples/messagebar

**Comment by betrue-final-final** [comment]
Ok, after this meeting

**Comment by betrue-final-final** [comment]
Why is this different than any other button in terms of focus? That cyan will not work against white no matter how thick it is. This should really be an inset black box inside just like standalone button.
image

**Comment by joschect** [comment]
@cliffkoh Have you addressed Ben's comments? is this ready to checkin?

**Comment by cliffkoh** [comment]
@betrue-final-final I used the hover styles I see in some other components (and used the system color Highlight) but your suggestion makes sense particularly if there is an existing buttons for High Contrast buttons to be treated this way. Happy to make the change (after hours).

**Comment by cliffkoh** [comment]
@betrue-final-final regarding the color, shouldn't it be the System color of "ActiveBorder" rather than whatever results in a white border like "ButtonText"?

I realize as well that the existing styles do not set an inset border.. in the Button component itself, the focus border has an outline offset of about 2px.

I will make it inset as you requested, but I will use what seems to be the more semantically correct color of "ActiveBorder" instead of "ButtonText" (which now makes me think the "Highlight" color that I see used in so many other components is wrong :\)

**Comment by cliffkoh** [comment]
Thank you @Vitalius1 and @betrue-final-final for in-depth review and valuable feedback. Hopefully this is now good to go in (and hopefully the use of "ActiveBorder" is non-contentious).

**Comment by cnamal** [comment]
hi @cliffkoh, any updates on this PR? :)

**Comment by cliffkoh** [comment]
I'm waiting for @betrue-final-final to approve the PR

**Comment by betrue-final-final** [comment]
Buttons look fine:
image

But the caption and expand button focus is almost invisible. I think there's a knock on effect here:
image

**Comment by cliffkoh** [comment]
@betrue-final-final what would you propose for caption and expand?

**Comment by cliffkoh** [comment]
Ping @betrue-final-final

**Comment by cnamal** [comment]
hey @cliffkoh, @betrue-final-final 
apologies for putting pressure, but we are starting to get close to SLA :(. Any updates on this PR?

**Comment by cliffkoh** [comment]
@dzearing @betrue-final-final can I merge this?

**Comment by cliffkoh** [comment]
I propose merging this right now and if there are any additional changes required, we can track them in a new issue. The change as-is does address the core issue reported and make concrete progress on the component's accessibility.

**Comment by cliffkoh** [comment]
image

Updated to use Window (same as the foreground text) so it looks like this. This only works because I also fixed (by ovewriting) the border to be inset instead of offset. Without the overwrite, the top and bottom portions of the border will not be visible.

**Comment by ecraig12345** [comment]
I'm technically the codeowner of MessageBar but don't have a strong opinion on this issue, so what @cliffkoh said seems fine. 

Also, the build is failing due to missing snapshot updates.

**Pull request merged**